plasmapy/utils/tests/test_code_repr.py

```diff
@@ -0,0 +1,370 @@
+        )
+
+
+@pytest.mark.parametrize("not_a_quantity", [3.5, "1.2"])
+def test__code_repr_of_quantity_typeerror(not_a_quantity):
+    """
+    Test that `_code_repr_of_quantity` raises a `TypeError` when
+    supplied with an object other than a `astropy.units.Quantity`.
+    """
+    with pytest.raises(TypeError):
+        _code_repr_of_quantity(not_a_quantity)
+
+
+def test__string_together_warnings_for_printing():
+    """Test that warning names and messages get strung together correctly."""
+    warnings = [UserWarning, DeprecationWarning]
+    warning_messages = ["msg1", "msg2"]
+    expected = "UserWarning: msg1\n\nDeprecationWarning: msg2"
+    actual = _string_together_warnings_for_printing(warnings, warning_messages)
+    if actual != expected:
+        pytest.fail(
+            f"_string_together_warnings_for_printing is not producing the "
+            f"expected result:\n"
+            f"  expected: {repr(expected)}\n"
+            f"  actual: {repr(actual)}"
+        )
+
+
+@pytest.mark.parametrize(
+    "arg, expected",
+    [
+        (1, "1"),
+        ("asdf", "'asdf'"),
+        (3.42, "3.42"),
+        ([3.42, 3.84], "[3.42, 3.84]"),
+        (4.2 * u.m, "4.2*u.m"),
+        (np.array([1, 2, 3]), "np.array([1, 2, 3])"),
+        (np.array([[1, 2], [3, 4]]), "np.array([[1, 2], [3, 4]])"),
+        (np.array([1.0, 2.0, 3.0]) * u.m / u.s, "np.array([1., 2., 3.])*u.m/u.s"),
+    ],
+)
+def test__code_repr_of_arg(arg, expected):
+    """
+    Test that _code_repr_of_arg correctly transforms arguments into a
+    `str` the represents how the arg would appear in code.
+    """
+    code_repr_of_arg = _code_repr_of_arg(arg)
+    if code_repr_of_arg != expected:
+        pytest.fail(
+            f"_code_repr_of_arg is not returning the expected result.\n"
+            f"arg = {arg}\n"
+            f"expected:         {expected}\n"
+            f"code_repr_of_arg: {code_repr_of_arg}"
+        )
+
+
+@pytest.mark.parametrize(
+    "args, kwargs, expected",
+    [
+        ((), {}, ""),
+        (1, {"a": "b"}, "1, a='b'"),
+        ((1, 2, 3), {"a": "A", "b": "B"}, "1, 2, 3, a='A', b='B'"),
+        (np.array([1.0, 2.0]), {"k": 3}, "np.array([1., 2.]), k=3"),
+        (np.array([1.0, 2.0]) * u.m / u.s, {}, "np.array([1., 2.])*u.m/u.s"),
+    ],
+)
+def test__code_repr_of_args_and_kwargs(args, kwargs, expected):
+    """
+    Test that `_code_repr_of_args_and_kwargs` returns a string containing
+    the positional and keyword arguments, as they would appear in a
+    function call.
+    """
+    args_and_kwargs = _code_repr_of_args_and_kwargs(args, kwargs)
+    if args_and_kwargs != expected:
+        pytest.fail(
+            f"_code_repr_of_args_and_kwargs with the following arguments:\n"
+            f"  args:   {args}\n"
+            f"  kwargs: {kwargs}\n"
+            f"is not returning the expected string:\n"
+            f"  expected: {repr(expected)}\n"
+            f"  actual:   {repr(args_and_kwargs)}"
+        )
```

Comment:
Testing and reporting failures this way seems like it's sidestepping the [`assert` mechanics](https://docs.python.org/3/reference/simple_stmts.html#the-assert-statement).  Why not do something more like...

```suggestion
    args_and_kwargs = _code_repr_of_args_and_kwargs(args, kwargs)
    assert args_and_kwargs == expected, (
        f"_code_repr_of_args_and_kwargs with the following arguments:\n"
        f"  args:   {args}\n"
        f"  kwargs: {kwargs}\n"
        f"is not returning the expected string:\n"
        f"  expected: {repr(expected)}\n"
    )
```

Reply:
I do think this approach should be applied to all similar cases.